tests/core/string/test_string.h

```diff
@@ -1988,6 +1988,48 @@ TEST_CASE("[String] Variant ptr indexed set") {
 	CHECK_EQ(s, String("azcd"));
 }
 
+TEST_CASE("[String][URL] Parse URL") {
+#define CHECK_URL(m_url_to_parse, m_expected_schema, m_expected_host, m_expected_port, m_expected_path, m_expected_error) \
+	do {                                                                                                                  \
+		int port;                                                                                                         \
+		String url(m_url_to_parse), schema, host, path;                                                                   \
+                                                                                                                          \
+		CHECK_EQ(m_expected_error, url.parse_url(schema, host, port, path));                                              \
+		CHECK_EQ(m_expected_schema, schema);                                                                              \
+		CHECK_EQ(m_expected_host, host);                                                                                  \
+		CHECK_EQ(m_expected_path, path);                                                                                  \
+		CHECK_EQ(m_expected_port, port);                                                                                  \
+	} while (false)
```

Comment:
We tend to use this approach for function-like macros instead:

https://github.com/godotengine/godot/blob/e3213aaef5e0e72b8272e65d989d3d8222be17ca/core/error/error_macros.h#L669-L677

See https://github.com/godotengine/godot/pull/60613 as for why this is used over `do { ... } while (false)`.